src/colorizer/utils/correlation.ts

```diff
@@ -0,0 +1,102 @@
+/**
+ * Calculates the Pearson correlation coefficient (a measurement of linear
+ * correlation) between two arrays of numbers. See
+ * https://en.wikipedia.org/wiki/Pearson_correlation_coefficient.
+ * @returns
+ * A number between -1 and 1 representing the correlation. Returns 0 if there
+ * are no finite values in the input arrays.
+ */
+export function pearson(x: Float32Array, y: Float32Array): number {
+  // Adapted from: https://memory.psych.mun.ca/tech/js/correlation.shtml
+  if (x.length !== y.length) {
+    throw new Error(
+      "correlations.ts/pearson: Data arrays must have the same length when calculating correlation coefficient."
+    );
+  }
+
+  const arrayLength = x.length;
+  const xy = new Float32Array(arrayLength);
+  const x2 = new Float32Array(arrayLength);
+  const y2 = new Float32Array(arrayLength);
```

Comment:
a future optimization would be to not create these three each time you call this function - they can be reused by the higher level loop

Reply:
Will add a TODO comment!